_posts/2022-12-16-aave-rebasing.md

```diff
@@ -0,0 +1,332 @@
+
+This is called _rebasing_.
+Said another way, aTokens are _rebase tokens_.
+
+Aave has a really interesting implementation of rebasing -- one which we believe is not well understood.
+One of the things we want to do in this post is explain how it works.
+
+#### Liquidity Indexes
+
+A key concept in understanding aToken rebasing is the notion of a __liquidity index__.
+It's defined in their [v2
+whitepaper](https://raw.githubusercontent.com/aave/protocol-v2/master/aave-v2-whitepaper.pdf) as follows:
+
+> Interest cumulated by the reserve during the time interval ∆T, updated
+> whenever a borrow, deposit, repay,
+> redeem, swap, liquidation event occurs.
+
+In other words, __the liquidity index for a certain asset is the total amount of interest earned by
+deposits of that asset into Aave__ -- expressed as a percentage scaled up by a
+[ray](https://github.com/aave/aave-v3-core/blob/f3e037b3638e3b7c98f0c09c56c5efde54f7c5d2/contracts/protocol/libraries/math/WadRayMath.sol#L17).
+
+Let's unpack that for a second, because it's a mouthful and this is really important.
+
+aPolyDAI currently has a liquidity index of `1010832786421347125844242759`, or
+1.083%. This means that _in aggregate_, over the entire lifetime of Aave, DAI deposits have earned 1.083%.
+So if you were the very first person to deposit DAI into
+Aave and you haven't touched it since, your deposit would today be worth 1.083% more.
+
+Each asset has an independent liquidity index associated with it.
+And indexes are constantly increasing.
+As noted in the whitepaper, an index increases any time a fee-generating action
+is taken by a user.
+
+Take borrowing, for instance.
+Suppose you've supplied DAI to Aave and someone else borrows DAI.
+The borrower will pay a certain amount of interest into Aave for the privilege, and that interest will accrue to you and other suppliers of DAI.
+
+_How_ it does so is by increasing the DAI liquidity index.
+It happens
+[here](https://github.com/aave/aave-v3-core/blob/f3e037b3638e3b7c98f0c09c56c5efde54f7c5d2/contracts/protocol/libraries/logic/ReserveLogic.sol#L291-L297) in the code:
+
+{% highlight js %}
+// Calculate the amount of interest that accrued since the last update.
+// The amount will be a ray-based percentage.
+uint256 cumulatedLiquidityInterest = MathUtils.calculateLinearInterest(
+  reserveCache.currLiquidityRate,
+  reserveCache.reserveLastUpdateTimestamp
+);
+
+// Add that interest to the current index by multiplying them together.
+reserveCache.nextLiquidityIndex = cumulatedLiquidityInterest.rayMul(
+  reserveCache.currLiquidityIndex
+);
+{% endhighlight %}
+
+For example, if the current liquidity index is `1050000000000000000000000000`
+(i.e. 5%) and another 2.5% has accrued since the last update, the new index will
+be `1076250000000000000000000000` (because 1.05 * 1.025 = 1.07625).
+
+It might not be obvious, but an increase in the liquidity index _just is_ the accrual of interest within Aave.
+The former constitutes the latter.
```

Comment:
Nit, but these two sentence were a bit confusing. Google says "constitutes" means "be a part of a whole" but really the two are identical?
```suggestion
It might not be obvious, but an increase in the liquidity index is identical to the accrual of interest within Aave.
```

Reply:
Huh, that's surprising.

I was using what I believe is it's [main definition](https://www.merriam-webster.com/dictionary/constitute), namely "makes up". Other ways of saying the same thing: accrual of interest is _reducible to_, is _nothing over and above_, occurs _solely in virtue of_ increases to the liquidity index.

Do you find any of those other idioms less confusing?

Reply:
Haha, to clarify it was a comment about how I thought "liquidity index == interest accrual" and not "liquidity index is one portion of interest accrual". Because the liquidity index tells you exactly how much interest accrued. So if "constitutes" is correct and "liquidity index == interest accrual" is incorrect, what are the other parts that make up interest accrual?

Reply:
> There are no other parts. "Be part of a whole" isn't the meaning of "constitutes" I was using. Google gave you a secondary meaning of "constitutes". The primary meaning isn't related to parts/wholes.

Oh that's a confusing set of definitions haha. Ok sounds good to me, consider this resolved then